packages/modsecurity/data_stream/auditlog/elasticsearch/ingest_pipeline/apache-modsec.yml

```
@@ -0,0 +1,112 @@
+      - set:
+          field: _temps.url
+          if: ctx.json.transaction.local_port == 80
+          value: "http://{{json.request.headers.Host}}:{{json.transaction.#local_port}}{{url.original}}"      
+      - uri_parts:
+          field: _temps.url
+          ignore_failure: true
+          keep_original: true
+          remove_if_successful: true         
+
+      - rename:
+          field: json.response.status
+          target_field: http.response.status_code
+          ignore_missing: true
+      - rename:
+          field: json.response.headers.Content-Type
+          target_field: http.response.mime_type
+          ignore_missing: true
+      - rename:
+          field: json.response.headers.Content-Length
+          target_field: http.response.bytes
+          ignore_missing: true
+      - convert:
+          field: http.response.bytes
+          type: long
+
+      - script:
+          lang: painless
+          source: >-
+                ctx._temps.result = ctx.json.audit_data.messages[0]
+      - grok:
+          field: _temps.result
+          patterns: 
+            - '%{GREEDYDATA} \[file "%{GREEDYDATA:modsec.audit.details.file}"\] \[line "%{GREEDYDATA:_temps.line_number}"\] \[id "%{GREEDYDATA:modsec.audit.details.rule_id}"\] \[msg "%{GREEDYDATA:modsec.audit.details.msg}"\] \[data "%{GREEDYDATA:modsec.audit.details.match}"\] \[severity "%{GREEDYDATA:modsec.audit.details.severity}"\] \[ver "%{DATA:modsec.audit.details.version}"\] %{GREEDYDATA:_temps.tags}'
+      - lowercase:
+          field: modsec.audit.details.severity
+          ignore_missing: true
+
+    # user agent and geoip enrich
+      - user_agent:
+          field: json.request.headers.User-Agent
+          ignore_missing: true
+      - geoip:
+          field: source.ip
+          target_field: source.geo
+          ignore_missing: true
+      - geoip:
+          database_file: GeoLite2-ASN.mmdb
+          field: source.ip
+          target_field: source.as
+          properties:
+          - asn
+          - organization_name
+          ignore_missing: true
+      - rename:
+          field: source.as.asn
+          target_field: source.as.number
+          ignore_missing: true
+      - rename:
+          field: source.as.organization_name
+          target_field: source.as.organization.name
+          ignore_missing: true
+      - set:
+          field: event.kind
+          value: event
+      - append:
+          field: event.category
+          value: web
+      - append:
+          field: event.type
+          value: access
+      - remove:
+          field: event.original
+          if: "ctx?.tags == null || !(ctx.tags.contains('preserve_original_event'))"
+          ignore_failure: true
+          ignore_missing: true
+      - remove:
+          field:
+            - json
+            - _temps
+          ignore_failure: true
+          ignore_missing: true             
```

```suggestion
          ignore_missing: true
```
Please ensure there is a final new line.